src/boundary_condition.cpp

```diff
@@ -278,10 +261,10 @@ void RotationalPeriodicBC::handle_particle(
   Direction u = p.u();
   double cos_theta = std::cos(theta);
   double sin_theta = std::sin(theta);
-  Position new_r = {
-    cos_theta * r.x - sin_theta * r.y, sin_theta * r.x + cos_theta * r.y, r.z};
-  Direction new_u = {
-    cos_theta * u.x - sin_theta * u.y, sin_theta * u.x + cos_theta * u.y, u.z};
+  Position new_r = {cos_theta * r[axis_1_idx_] - sin_theta * r[axis_2_idx_],
+    sin_theta * r[axis_1_idx_] + cos_theta * r[axis_2_idx_], r[zero_axis_idx]};
+  Direction new_u = {cos_theta * u[axis_1_idx_] - sin_theta * u[axis_2_idx_],
+    sin_theta * u[axis_1_idx_] + cos_theta * u[axis_2_idx_], u[zero_axis_idx]};
```

Comment:
Just realized that my "generalization" was based on the z-case that existed here before. The x and z cases should be similar but the y rotation matrix has the sign flipped, which I think explains why I'm seeing one test failure in this way. If we condition on `zero_axis_idx` I think I can get it right
<img width="354" height="327" alt="rotation_matrices" src="https://github.com/user-attachments/assets/79f73257-93d5-435f-a108-1933274da3c9" />